ext/srt/gstsrtsrc.c

```diff
@@ -286,6 +310,20 @@ gst_srt_src_class_init (GstSRTSrcClass * klass)
       g_signal_new ("caller-rejected", G_TYPE_FROM_CLASS (klass),
       G_SIGNAL_RUN_LAST, 0, NULL, NULL, NULL, G_TYPE_NONE, 0);
 
+  /**
+   * GstSRTSrc::caller-connecting:
+   * @gstsrtsrc: the srtsrc element that emitted this signal
+   *
+   * Whether to accept or reject a caller's connection to srtsrc in listener mode.
+   * The Caller's connection is rejected if the callback returns FALSE, else
+   * the connection is accepeted.
+   */
+  signals[SIG_CALLER_CONNECTING] =
+      g_signal_new_class_handler ("caller-connecting", G_TYPE_FROM_CLASS (klass),
+      G_SIGNAL_RUN_LAST, G_CALLBACK (src_caller_connecting),
```

Comment:
Rather than adding a static callback here, it should be a virtual function and let a user be able to override it.

Reply:
Changed static callback.